datafusion/physical-plan/src/spill/mod.rs

```diff
@@ -35,7 +35,10 @@ use datafusion_common::{exec_datafusion_err, HashSet, Result};
 
 fn read_spill(sender: Sender<Result<RecordBatch>>, path: &Path) -> Result<()> {
     let file = BufReader::new(File::open(path)?);
-    let reader = StreamReader::try_new(file, None)?;
+    // SAFETY: DataFusion's spill writer strictly follows Arrow IPC specifications
+    // with validated schemas and buffers. Skip redundant validation during read
+    // to speedup read operation. This is a deliberate safety-performance tradeoff.
```

Comment:
```suggestion
    // to speedup read operation. This is safe for DataFusion as input guarenteed to be correct when written.
```
I don't think it's really a tradeoff in the sense that we are losing safety.